airflow/providers/common/sql/hooks/sql.py

```diff
@@ -102,13 +102,13 @@ class DbApiHook(BaseForDbApiHook):
     """
 
     # Override to provide the connection name.
-    conn_name_attr = None  # type: str
+    conn_name_attr: str | None = None
```

Comment:
```suggestion
    conn_name_attr: str
```

We don’t need to None case (and this gets rid of the `cast` calls below).

Reply:
I did that because it was clashing with the definition of this class attribute in `BaseDatabricksHook` and  `QuboleHook`. (when inheriting from both of them for instance in `QuboleCheckHook`, or `DatabricksSqlHook`)

But you are right, it should be a 'str' here and also in `BaseDatabricksHook` and  `QuboleHook`.

I just updated that.